x-pack/legacy/plugins/infra/common/http_api/metrics_explorer/index.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import * as rt from 'io-ts';
+
+export const metricsExplorerAggregationRT = rt.keyof({
```

Comment:
I wonder if we should create an array for each of these aggregations
```
const METRIC_EXPLORER_AGGREGATIONS = ['avg', 'max', 'min', 'cardinality', 'rate', 'count'];
```
then define `metricsExplorerAggregagtionRT` as
```
export const metricsExplorerAggregationRT = rt.keyof(
  METRIC_EXPLORER_AGGREGATIONS.reduce(
  (acc, agg) => ({ ...acc, [agg]: null })
  , {})
);
```
Then in x-pack/legacy/plugins/infra/public/components/metrics_explorer/aggregation.tsx we can just include `METRICS_EXPLORER_AGGREGATIONS` and use it on [line 68](https://github.com/elastic/kibana/pull/51615/files#diff-e5ba648c4dc9fdcb312a0ba814e0bb0dR68)

Reply:
That reduce might not work, I think io-ts needs that to be a literal object instead of a dynamic one.